test/async-hooks/test-timerwrap.setTimeout.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const tick = require('./tick');
+const initHooks = require('./init-hooks');
+const { checkInvocations } = require('./hook-checks');
+const TIMEOUT = common.platformTimeout(100);
+
+const hooks = initHooks();
+hooks.enable();
+
+// install first timeout
+setTimeout(common.mustCall(ontimeout), TIMEOUT);
+const as = hooks.activitiesOfTypes('TIMERWRAP');
+assert.strictEqual(as.length, 1, 'one timer wrap when first timeout installed');
+const t1 = as[0];
+assert.strictEqual(t1.type, 'TIMERWRAP', 'timer wrap');
+assert.strictEqual(typeof t1.uid, 'number', 'uid is a number');
+assert.strictEqual(typeof t1.triggerId, 'number', 'triggerId is a number');
+checkInvocations(t1, { init: 1 }, 't1: when first timer installed');
+
+function ontimeout() {
+  checkInvocations(t1, { init: 1, before: 1 }, 't1: when first timer fired');
+
+  // install second timeout with same TIMEOUT to see timer wrap being reused
```

Comment:
This comment leads me to believe that /bcbc6b03a59bac38a5128d1429faac2c489e304a may be incorrect.

I don't think there should be anything actually timeout duration based in this test since it deals with inheritance. That should be deterministic.